generated/nidcpower/nidcpower/session.py

```diff
@@ -4795,17 +4796,36 @@ def fetch_multiple(self, count, timeout=hightime.timedelta(seconds=1.0)):
         Returns:
             measurements (list of Measurement): List of named tuples with fields:
 
+                - **channel** (str)
                 - **voltage** (float)
                 - **current** (float)
                 - **in_compliance** (bool)
 
         '''
         import collections
-        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance'])
+        Measurement = collections.namedtuple('Measurement', ['channel', 'voltage', 'current', 'in_compliance'])
 
-        voltage_measurements, current_measurements, in_compliance = self._fetch_multiple(timeout, count)
+        voltage_measurements, current_measurements, in_compliances = self._fetch_multiple(timeout, count)
 
-        return [Measurement(voltage=voltage_measurements[i], current=current_measurements[i], in_compliance=in_compliance[i]) for i in range(count)]
+        if self._repeated_capability == '':
+            channel_names = self._get_channel_names(range(self.channel_count))
+        else:
+            first_channel_name = self._get_channel_names([0])[0]
```

Comment:
Need a comment on why `first_channel_name` is needed; which as I understand is to handle the case where `channels` is something like `Dev1/1:2,0`.

(Commentary: I think we made a mistake by allowing users to pass channels with mixed formatting (channel indices and FQCNs).

Reply:
I guess there is also the case where the resource_name is `dev1/0:3` and repeated capability is `0`.

Reply:
I have added a comment to explain why `first_channel_name` is needed in my latest commit.

```
            # first_channel_name is used to deduce if the session was opened with independent_channels
            #  set to True (in that case, the channel name would be fully-qualified) or False (in
            #  that case, the channel name would not have any instrument prefix). It is also used
            #  to extract the instrument prefix and populate the expanded channels repeated
            #  capabilities with it if needed
```